hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/state/ContractStateStore.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.contract.impl.state;
+
+import com.hedera.hapi.node.state.common.EntityNumber;
+import com.hedera.hapi.node.state.contract.Bytecode;
+import com.hedera.hapi.node.state.contract.SlotKey;
+import com.hedera.hapi.node.state.contract.SlotValue;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Set;
+
+/**
+ * An intermediary that manages access to the slot and bytecode key/value states.
+ */
+public interface ContractStateStore {
+    /**
+     * Returns the {@link Bytecode} for the given contract number.
+     *
+     * @param contractNumber the contract number to get the {@link Bytecode} for
+     * @return the {@link Bytecode} for the given contract number
+     */
+    Bytecode getBytecode(@NonNull EntityNumber contractNumber);
```

Comment:
I notice `SlotKey` (a BESU function) takes the contract as a `long`, IOW the EVM address.  Here we're using a Hedera `EntityNumber`.  Will callers have to go to any trouble here, that is, will they have the evm address and have to translate it themselves?  If so, maybe a default method could be here to do the translation.

Reply:
Hmm I'm not quite sure. It might be hard to know until the `HandleContext` API is nailed down and we've sorted out exactly which of these interfaces even need to exist. 🤔